src/TxPool/Txpool.ts

```diff
@@ -0,0 +1,135 @@
+import Config from '../Config';
+import { Pool, ITxPoolTxDetails } from './Pool';
+import W3Util from '../Util';
+import Cache, { ICachedTxDetails } from '../Cache';
+
+const SCAN_INTERVAL = 5000;
+
+export default class TxPool {
+    public config: Config;
+    public pool: any = [];
+    public subs: any = {};
+  
+    constructor(config: Config) {
+      this.config = config;
+      this.pool = new Pool();
+    }
+
+    public get cache (): Cache<ICachedTxDetails> {
+        return this.config.cache;
+    }
+
+    public get logger (): any {
+        return this.config.logger;
+    }
+
+    public get util (): W3Util {
```

Comment:
???

Reply:
use getter to retrieve `logger` and `util`, reduces the chances of memory leakage and misreferenced object and is supposedly cheaper.